### ISSUE TITLE
Guidance on how to pick slices and max/min inputs?

### ISSUE THREAD

**Issue by wodano** [animetosho/ParPar#30]
As a noob, I don't really know the ideal way of picking correct amount of slices, is there a guide on how to pick? So far I've just been doing random numbers with:

~/ParPar-0.3.2/bin/parpar.js -s random$M -r12% -o input ~/input.mkv 

Sorry if this is really basic, but I haven't seen other people ask this question on Reddit or elsewhere.

Edit: I also meant to ask how slices should be adjusted according to the size and amount of the file(s).

**Comment by animetosho** [comment]
Yeah, it's unfortunate that it's a required parameter that can't be auto determined, and picking an appropriate size isn't always straightforward.

At a basic level, PAR2 splits the input file into slices, which is what recovery is computed from (if there's more than one input file, it's effectively joined into one big file). The slice size allows you to adjust how the splitting is performed - if you choose to have a lot of slices, then each slice must be smaller than if you chose to have fewer slices.  
Note that I refer to the above as "input slices", as it's slicing the input data. The PAR2 creation takes these input slices, and generates a number of recovery slices (where the slice size for both input and recovery slices must be identical), which is stored in the output PAR2 files.

Hopefully the following points gives you some idea of how to pick a size:

* in general, more recovery slices improves recoverability/flexibility, but is slower to process. If one byte of data becomes corrupt, the entire input slice the byte is in, is considered to be corrupt and needs repair. So bigger slices means that errors effectively affect more of the input data, but are faster to compute due to there being less slices to manage.
* PAR2 imposes a maximum of 32768 input slices, so whatever you pick, you can't go beyond that figure
* where PAR2 is being used may play a role too.  
For example, if you're uploading files to Usenet, using a 700KB article size, then it makes little sense to have a slice size smaller than 700KB, since a missing article means that 700KB of data is effectively lost. Similarly, you may want to keep the slice size to a multiple of 700KB, if larger slices are desired/needed.

To sum it up, consider the following:

* if the underlying medium has some block size, keep the slice size to a multiple of this figure
* choose where you want to be on the speed vs recoverability line

**Comment by wodano** [comment]
Thank you so much for taking the time to write all of that!

> where PAR2 is being used may play a role too.

I'm so glad I asked now, my slice sizes have been larger than 700KB which I see is an issue.
Anyway, thanks for the help and the pizza analogy was a nice way to describe it.
I think I'll be going with a middle ground of speed vs recoverability.
Thanks!
